test/upgrade/suite.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"knative.dev/eventing-kafka/test/upgrade/installation"
+	pkgupgrade "knative.dev/pkg/test/upgrade"
+)
+
+// Suite defines the whole upgrade test suite for Eventing Kafka.
+func Suite() pkgupgrade.Suite {
+	return pkgupgrade.Suite{
+		Tests: pkgupgrade.Tests{
+			PreUpgrade: []pkgupgrade.Operation{
+				ChannelPreUpgradeTest(),
+			},
+			PostUpgrade: []pkgupgrade.Operation{
+				ChannelPostUpgradeTest(),
+			},
+			PostDowngrade: []pkgupgrade.Operation{
```

Comment:
for these. somewhat possible to make better usage of "upstream" (eventing)? 

looks like very similar over there

Reply:
Once we have the source also covered.


will we register new functions there?  Or will be rename to more generat.... like just `PostDowngradeTest` ?

Reply:
What do you mean actually? 

The framework is from `knative.dev/pkg` but those operations are completely different here and in eventing core.

Reply:
For source it will be something like:

```golang
PostUpgrade: []pkgupgrade.Operation{
  ChannelPostUpgradeTest(),
  SourcePostUpgradeTest(),
},
```

And we can even introduce some logic to conditionally build this test suite.